web/api/v1/api.go

```diff
@@ -889,6 +893,50 @@ func (api *API) serveFlags(r *http.Request) apiFuncResult {
 	return apiFuncResult{api.flagsMap, nil, nil, nil}
 }
 
+// Stat holds the information about individual cardinality.
+type stat struct {
+	Name  string `json:"name"`
+	Value uint64 `json:"value"`
+}
+
+// tsdbStatus has information of cardinality statistics from postings.
+type tsdbStatus struct {
+	QueryDuration               string `json:"queryTime"`
+	SeriesCountByMetricName     []stat `json:"seriesCountByMetricName"`
+	LabelValueCountByLabelName  []stat `json:"labelValueCountByLabelName"`
+	MemoryInBytesByLabelName    []stat `json:"memoryInBytesByLabelName"`
+	SeriesCountByLabelValuePair []stat `json:"seriesCountByLabelValuePair"`
+}
+
+func (api *API) serveTSDBStatus(r *http.Request) apiFuncResult {
+	db := api.db()
+	if db == nil {
+		return apiFuncResult{nil, &apiError{errorUnavailable, errors.New("TSDB not ready")}, nil, nil}
+	}
+	conver := func(stats []index.Stat) []stat {
+		result := make([]stat, 0, len(stats))
+		for _, item := range stats {
+			item := stat{Name: item.Name, Value: item.Count}
+			result = append(result, item)
+		}
+		return result
+	}
+
+	startTime := time.Now()
+	posting := db.Head().PostingsCardinalityStats(model.MetricNameLabel)
+	totalQueryTime := fmt.Sprintf("%.3f", time.Since(startTime).Seconds())
+
+	response := tsdbStatus{
+		QueryDuration:               totalQueryTime,
```

Comment:
I don't think we need this anymore. Now that it is its own endpoint/page we can just use the built in Prometheus metrics.

Reply:
Clarification, I specifically do not think we need to return QueryDuration since it is now a proper route and api call and will have other metrics around it.

Reply:
cool

Reply:
Ah yeah, good point @csmarchbanks.